org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/OrganizeImportsHandler.java

```diff
@@ -112,104 +96,16 @@ public static TextEdit organizeImports(ICompilationUnit unit, Function<ImportSel
 		try {
 			JobHelpers.waitForJobs(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS, monitor);
 			TextEdit edit = op.createTextEdit(null);
-			// See https://bugs.eclipse.org/bugs/show_bug.cgi?id=283287
-			// and https://github.com/redhat-developer/vscode-java/issues/1472
-			TextEdit staticEdit = wrapStaticImports(edit, astRoot, unit);
-			if (staticEdit.getChildrenSize() == 0) {
+			if (edit instanceof MultiTextEdit && edit.getOffset() == 0 && edit.getLength() == 0 && ((MultiTextEdit) edit).getChildrenSize() == 0) {
 				return null;
 			}
-			return staticEdit;
+			return edit;
 		} catch (OperationCanceledException | CoreException e) {
 			JavaLanguageServerPlugin.logException("Failed to resolve organize imports source action", e);
 		}
 		return null;
 	}
 
-	public static TextEdit wrapStaticImports(TextEdit edit, CompilationUnit root, ICompilationUnit unit) throws MalformedTreeException, CoreException {

```

Comment:
This is being removed because the fix occured upstream, right ? I guess this code was also want was causing the removal ?

Reply:
Right.